staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go

```diff
@@ -99,6 +122,9 @@ func NestedInt64(obj map[string]interface{}, fields ...string) (int64, bool) {
 		return 0, false
 	}
 	i, ok := val.(int64)
+	if !ok && invalidTypeDetectionEnabled {
```

Comment:
My understanding of the bool in the return values is to allow this situation. It's a bit strange to panic.